build_tools/fetch_sources.py

```diff
@@ -100,81 +76,97 @@ def run(args):
+        log(f"ERROR: Patch directory {patch_version_dir} does not exist")
     for patch_project_dir in patch_version_dir.iterdir():
-        print(f"* Processing project patch directory {patch_project_dir}:")
-        project_dir: Path = args.dir / patch_project_dir.name
+        log(f"* Processing project patch directory {patch_project_dir}:")
+        submodule_path = get_submodule_path(patch_project_dir.name)
+        project_dir = THEROCK_DIR / submodule_path
         if not project_dir.exists():
-            print(f"WARNING: Source directory {project_dir} does not exist. Skipping.")
+            log(f"WARNING: Source directory {project_dir} does not exist. Skipping.")
             continue
         patch_files = list(patch_project_dir.glob("*.patch"))
         patch_files.sort()
-        print(f"Applying {len(patch_files)} patches")
+        log(f"Applying {len(patch_files)} patches")
         exec(["git", "am", "--whitespace=nowarn"] + patch_files, cwd=project_dir)
+        # Since it is in a patched state, make it invisible to changes.
+        exec(
+            ["git", "update-index", "--skip-worktree", "--", submodule_path],
+            cwd=THEROCK_DIR,
+        )
+
+
+# Gets the the relative path to a submodule given its name.
+# Raises an exception on failure.
+def get_submodule_path(name: str) -> str:
+    relpath = (
+        subprocess.check_output(
+            [
+                "git",
+                "config",
+                "--file",
+                ".gitmodules",
+                "--get",
+                f"submodule.{name}.path",
+            ],
+            cwd=str(THEROCK_DIR),
+        )
+        .decode()
+        .strip()
+    )
+    return relpath
 
 
 def populate_ancillary_sources(args):
     """Various subprojects have their own mechanisms for populating ancillary sources
     needed to build. There is often something in CMake that attempts to automate it,
     but it is also often broken. So we just do the right thing here as a transitionary
     step to fixing the underlying software packages."""
-    populate_submodules_if_exists(args, args.dir / "rocprofiler-register")
-    populate_submodules_if_exists(args, args.dir / "rocprofiler-sdk")
+    populate_submodules_if_exists(args, THEROCK_DIR / "base" / "rocprofiler-register")
+    populate_submodules_if_exists(args, THEROCK_DIR / "profiler" / "rocprofiler-sdk")
 
     # TODO(#36): Enable once rocprofiler-systems can be checked out on Windows
     #     error: invalid path 'src/counter_analysis_toolkit/scripts/sample_data/L2_RQSTS:ALL_DEMAND_REFERENCES.data.reads.stat'
     #  Upstream issues:
     #   https://github.com/ROCm/rocprofiler-systems/issues/105
     #   https://github.com/icl-utk-edu/papi/issues/321
     if not is_windows():
-        populate_submodules_if_exists(args, args.dir / "rocprofiler-systems")
+        populate_submodules_if_exists(
+            args, THEROCK_DIR / "profiler" / "rocprofiler-systems"
+        )
```

Comment:
Glad this is still working after switching from `repo` to submodules. papi merged a patch to fix the checkout on Windows but I haven't received a response yet on https://github.com/ROCm/rocprofiler-systems/issues/105 to update the papi version pin in rocprofiler-systems